src/fomo.jl

```diff
@@ -0,0 +1,419 @@
+export fomo, FomoSolver, FoSolver, R2, TR, tr_step, r2_step
```

Comment:
I feel one thing is just sligthly strange, calling `R2` with `FoSolver` instead of `R2Solver`, just from the naming perspective,  is a bit weird.
What about keeping `R2Solver` for now and use `Base.@deprecate`?
Or if that's make sense `FoSolver` should be called `R2Solver`?

Reply:
I totally agree that we should make better use of `Base.@deprecate`.

Reply:
In the end I see 3 solutions:
- Remove `TR` and `fo` methods, leave only `R2` and rename `FoSolver` into `R2Solver`
- Remove `TR` and `R2` methods, leave only `fo` and deprecate `R2` and `R2Solver`
- Leave the 3 methods `fo`, `TR` and `R2` as it is, and deprecate `R2Solver`.

Reply:
It seems to me the minimal change is to just deprecate `R2Solver`?
We can still keep the name `R2` for the method as it is used in the papers.